File: server/src/main/java/dietcourtserver/model/MenuID.java
package dietcourtserver.model;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Objects;

@Embeddable
public class MenuID
        implements Serializable {

    @Column(name = "UserID")
    private int userID;

    @Column(name = "DishID")
    private int dishID;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;

        if (o == null || getClass() != o.getClass())
            return false;

        MenuID that = (MenuID) o;
        return Objects.equals(userID, that.userID) &&
                Objects.equals(dishID, that.dishID);
    }

    @Override
    public int hashCode() {
        return Objects.hash(userID, dishID);
    }
}

File: client/app/src/main/java/com/etu/client/data/model/MenuID.kt
package com.etu.client.data.model

import java.io.Serializable
import java.util.*

class MenuID(userID: Int, dishID: Int) : Serializable {

    private val userID = 0
    private val dishID = 0

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as MenuID
        return userID == that.userID &&
                dishID == that.dishID
    }

    override fun hashCode(): Int {
        return Objects.hash(userID, dishID)
    }
}